Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Sync/DataSynchronizer.swift

```diff
@@ -327,8 +329,13 @@ public class DataSynchronizer: NetworkStateListener, FatalErrorListener {
      */
     func count(filter: Document,
                options: CountOptions?,
-               in namespace: MongoNamespace) {
-        fatalError("\(#function) not implemented")
+               in namespace: MongoNamespace) throws -> Int {
+        let lock = self.syncConfig[namespace]?.nsLock ?? syncLock
```

Comment:
I know subscripts can't throw, which is why I suggested for this method to throw.

Reply:
i.e. if the subscript returns nil, `throw` here instead of coalescing the nil value to `syncLock`

Reply:
a `StitchError.clientError` perhaps. It seems like it would be better for `count` to throw immediately rather than having it exhibit undefined behavior until the fatal error listener can react to the error and emit an error for the user to handle.